pkg/msr/msr_linux.go

```diff
@@ -8,20 +8,66 @@ package msr
 import (
 	"encoding/binary"
 	"fmt"
+	"io/ioutil"
 	"os"
 	"path/filepath"
 	"strconv"
+	"strings"
 )
 
 // CPUs is a slice of the various cpus to read or write the MSR to.
 type CPUs []uint64
 
+func parseCPUs(s string) (CPUs, error) {
+	cpus := make(CPUs, 0)
+	// We expect the format to be "0-5,7-8..." or we could also get just one cpu.
+	// We're unlikely to get more than one range since we're looking at present cpus,
+	// but handle it just in case.
+	ranges := strings.Split(strings.TrimSpace(s), ",")
+	for _, r := range ranges {
+		if len(r) == 0 {
+			continue
+		}
+		// Split on a - if it exists.
+		cs := strings.Split(r, "-")
+		switch len(cs) {
+		case 1:
+			u, err := strconv.ParseUint(cs[0], 0, 64)
+			if err != nil {
+				return nil, fmt.Errorf("unknown cpu range: %v, failed to parse %v", r, err)
+			}
+			cpus = append(cpus, uint64(u))
+		case 2:
+			ul, err := strconv.ParseUint(cs[0], 0, 64)
+			if err != nil {
+				return nil, fmt.Errorf("unknown cpu range: %v, failed to parse %v", r, err)
+			}
+			uh, err := strconv.ParseUint(cs[1], 0, 64)
+			if err != nil {
+				return nil, fmt.Errorf("unknown cpu range: %v, failed to parse %v", r, err)
+			}
+			for i := ul; i <= uh; i++ {
+				cpus = append(cpus, uint64(i))
```

Comment:
I think you need a map. People will do stuff like 0-4,2-5 and it will get messy, unless I'm missing something.

Reply:
we had ranges like this in bproc and it was amazing what people would type. segv followed, as it was C.

Reply:
I'll just sort and remove dupes